uber/automated_emails.py

```diff
@@ -1,75 +1,290 @@
-# WARNING - changing the email subject line for an email causes ALL of those emails to be re-sent!
-# Note that since c.EVENT_NAME is used in most of these emails, changing the event name mid-year
-# could cause literally thousands of emails to be re-sent!
-
 from uber.common import *
 
 
 class AutomatedEmail:
+    """
+    Represents one category of emails that we send out.
+
+    An example of an email category would be "Your registration has been confirmed".
+    Each category
+    """
+
+    # all instances of every registered email category in the system
     instances = OrderedDict()
 
+    # a list of queries to run during each automated email sending run to
+    # return particular model instances of a given type.
     queries = {
-        Attendee: lambda session: session.staffers(only_staffing=False),
+        Attendee: lambda session: session.all_attendees(),
         Group: lambda session: session.query(Group).options(subqueryload(Group.attendees))
     }
-    extra_models = queries  # we've renamed "extra_models" to "queries" but are temporarily keeping the old name for backwards-compatibility
 
-    def __init__(self, model, subject, template, filter, *, sender=None, extra_data=None, cc=None, bcc=None, post_con=False, needs_approval=True, ident=None):
-        self.model, self.template, self.needs_approval = model, template, needs_approval
+    def __init__(self, model, subject, template, filter, *, when=(),
+                 sender=None, extra_data=None, cc=None, bcc=None,
+                 post_con=False, needs_approval=True, ident=None):
+
         self.subject = subject.format(EVENT_NAME=c.EVENT_NAME)
+        self.ident = ident or self.subject
```

Comment:
I had a reason for writing this line the way I did. From the PR:
>I left this essentially as-is (though I did clean it up, as suggested) because the idea behind it is that you can change an email's subject by copying the old subject into the ident variable. We may want to undo the formatting later for the reason you gave, but since MAGLabs is up and running and then MAG2017 will be using the old format, it'll be a while.

If we're really okay with not being able to change any emails' subjects that have already been sent, we can leave it as-is. Otherwise, it should be reverted back to formatting both ident and subject.